test/performance/scale_from_zero_test.go

```diff
@@ -69,24 +70,31 @@ func runScaleFromZero(idx int, t *testing.T, clients *test.Clients, ro *test.Res
 
 	start := time.Now()
 	t.Logf("%02d: waiting for endpoint to serve request", idx)
-	if _, err := pkgTest.WaitForEndpointStateWithTimeout(
+	resp, err := pkgTest.WaitForEndpointStateWithTimeout(
 		clients.KubeClient,
 		t.Logf,
 		domain,
 		pkgTest.MatchesAllOf(pkgTest.IsStatusOK, pkgTest.MatchesBody(helloWorldExpectedOutput)),
 		"HelloWorldServesText",
-		test.ServingFlags.ResolvableDomain, waitToServe); err != nil {
+		test.ServingFlags.ResolvableDomain, waitToServe)
+	if err != nil {
 		m := fmt.Sprintf("%02d: the endpoint for Route %q at domain %q didn't serve the expected text %q: %v", idx, ro.Route.Name, domain, helloWorldExpectedOutput, err)
 		t.Log(m)
 		return 0, errors.New(m)
 	}
-
+	dur := time.Since(start)
 	t.Logf("%02d: request completed", idx)
-	return time.Since(start), nil
+
+	// Sleep to get traces
+	time.Sleep(5 * time.Second)
+	traceID := resp.Header.Get(zipkin.ZipkinTraceIDHeader)
+	AddTrace(t.Logf, t.Name(), traceID)
```

Comment:
Just to make sure I understand this part, this is adding the zipkin logging for all requests made from scale_from_zero... test. From my previous testing, for a response with 200 status code, 5 seconds is very flaky on when the trace result would be available. Were you able to get trace result consistently with 5 second delay?

Reply:
yes. I was able to get the traces consistently. Maybe there was some issue with your cluster. We are using a bigger cluster with more resources in the test cluster. We can evaluate this going forward and increase if needed